src/uiwidgets/ScriptsPane.as

```diff
@@ -420,21 +420,16 @@ return true; // xxx disable this check for now; it was causing confusion at Scra
 	}
 
 	private function dropCompatible(droppedBlock:Block, target:DisplayObject):Boolean {
-		const menusThatAcceptReporters:Array = [
-			'broadcast', 'costume', 'backdrop', 'scene', 'sound',
-			'spriteOnly', 'spriteOrMouse', 'location', 'spriteOrStage', 'touching'];
 		if (!droppedBlock.isReporter) return true; // dropping a command block
 		if (target is Block) {
 			if (Block(target).isEmbeddedInProcHat()) return false;
 			if (Block(target).isEmbeddedParameter()) return false;
+		} else {
+			if (BlockArg(target).type == 'h') return false;
 		}
 		var dropType:String = droppedBlock.type;
-		var targetType:String = target is Block ? Block(target.parent).argType(target).slice(1) : BlockArg(target).type;
-		if (targetType == 'm') {
-			if (Block(target.parent).type == 'h') return false;
-			return menusThatAcceptReporters.indexOf(BlockArg(target).menuName) > -1;
-		}
-		if (targetType == 'b') return dropType == 'b';
+		var targetType:String = (target is Block) ? Block(target).type : BlockArg(target).type;
```

Comment:
This looks quite simple compared to the previous code on this line. Surely there's a reason..

..and, in fact, there is: See the commit ebe20dfe8bdf293c507096d6c7d78b1b889b7e70.

>Allow dropping reporters on predicates in text/number slots

Have you tested to see if that still works, now? Or by making everything block compatible I guess it's just not necessary at all?